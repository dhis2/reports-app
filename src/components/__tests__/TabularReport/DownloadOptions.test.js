import React from 'react'
import { shallow } from 'enzyme'
import {
    DownloadOptions,
    DownloadOption,
} from '../../TabularReport/DownloadOptions'

describe('<DownloadOptions/>', () => {
    const fileUrls = [
        {
            extension: 'pdf',
            url: 'http://www.google.nl',
        },
        {
            extension: 'xls',
            url: 'http://www.google.com',
        },
    ]

    it('Renders an empty container when fileUrls is an empty array', () => {
        expect(shallow(<DownloadOptions fileUrls={[]} />)).toMatchSnapshot()
    })
    it('Renders an empty container when fileUrls is an empty array', () => {
        expect(
            shallow(<DownloadOptions fileUrls={fileUrls} />)
        ).toMatchSnapshot()
    })

    it('Will open PDF files in a new window (open in browser)', () => {
        const file = {
            extension: 'pdf',
            url: 'http://www.google.nl',
        }
        const wrapper = shallow(<DownloadOption file={file} id={0} />)
        const linkTarget = wrapper.find('a').prop('target')
        expect(linkTarget).toEqual('_blank')
    })

    it('Will open other files in the same window (download)', () => {
        const wrapper = shallow(<DownloadOption file={fileUrls[1]} id={1} />)
        const linkTarget = wrapper.find('a').prop('target')
        expect(linkTarget).toEqual('_self')
    })
})
