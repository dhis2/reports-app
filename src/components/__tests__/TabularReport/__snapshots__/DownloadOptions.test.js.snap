// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DownloadOptions/> Renders an empty container when fileUrls is an empty array 1`] = `
<div
  id="download-options-container"
  style={
    Object {
      "display": "flex",
      "justifyContent": "flex-end",
    }
  }
/>
`;

exports[`<DownloadOptions/> Renders an empty container when fileUrls is an empty array 2`] = `
<div
  id="download-options-container"
  style={
    Object {
      "display": "flex",
      "justifyContent": "flex-end",
    }
  }
>
  <Component
    file={
      Object {
        "extension": "pdf",
        "url": "http://www.google.nl",
      }
    }
    key="0"
  />
  <Component
    file={
      Object {
        "extension": "xls",
        "url": "http://www.google.com",
      }
    }
    key="1"
  />
</div>
`;
